docs/terraform.md

```diff
@@ -1,74 +1,145 @@
-## Building Kubernetes clusters with terraform
+## Building Kubernetes clusters with Terraform
 
-Kops can generate terraform configurations, and you can then apply them using the terraform plan/apply tools. 
-This is very handy if you are already using terraform, or if you want to check in the terraform output into
-version control.
+Kops can generate Terraform configurations, and then you can then apply them using the `terraform plan` and `terraform apply` tools. This is very handy if you are already using Terraform, or if you want to check in the Terraform output into version control.
 
-The terraform output should be reasonably stable (i.e. the text files should only change where something has actually
-changed - items should appear in the same order etc).
+The gist of it is that, instead of letting kops apply the changes, you tell kops what you want, and then kops spits out what it wants done into a `.tf` file. **_You_** are then responsible for turning those plans into reality.
 
+The Terraform output should be reasonably stable (i.e. the text files should only change where something has actually changed - items should appear in the same order etc). This is extremely useful when using version control as you can diff your changes easily.
 
-### Using terraform
+Note that if you modify the Terraform files that kops spits out, it will override your changes with the configuration state defined by it's own configs. In other terms, kops own state is the ultimate source of truth (as far as kops is concerned), and Terraform is a representation of that state for your convenience.
 
-To use terraform, you simply run update with `--target=terraform` (but see below for a workaround for a bug
-if you are using a terraform version before 0.7)
+Ps: Steps below assume a recent version of Terraform. There's a workaround for a bug if you are using a Terraform version before 0.7 that you should be aware of (see _"Caveats"_ section).
+
+### Using Terraform
+
+#### Set up remote state
+
+You could keep your Terraform state locally, but we recommend saving it on S3 (consider turning on versioning on that bucket), like so:
+
+```
+$ terraform remote config \
+  -backend=s3 \
+  -backend-config="bucket=mycompany.terraform>" \
+  -backend-config="key=sandbox/infrastructure.tfstate" \
+  -backend-config="region=us-east-1"
+```
+
+Learn more [about Terraform state here](https://www.terraform.io/docs/state/remote/index.html).
+
+#### Initialize/create a cluster
 
 For example, a complete setup might be:
 
 ```
-export KOPS_STATE_STORE=s3://<somes3bucket>
-export CLUSTER_NAME=<kubernetes.mydomain.com>
-${GOPATH}/bin/kops create cluster ${CLUSTER_NAME} --zones us-east-1c
-${GOPATH}/bin/kops update cluster ${CLUSTER_NAME} --target=terraform
+$ kops create cluster \
+  --name=sandbox.kubernetes.mycompany.tld \
```

Comment:
Most of the other docs use clusternames like `kubernetes.example.com` or similar. This comment is partially directed at you, but now that we have so many folks contributing to the docs... I think we should create a "docs contribution standards" so we can spell some of this stuff out and easily know if it fits with the rest or not.